crates/iota-indexer-streaming/src/memory.rs

```diff
@@ -409,3 +473,83 @@ impl InMemory {
         Ok(stored)
     }
 }
+
+/// A [`Stream`] wrapper that provides metrics capabilities for the
+/// [`BroadcastStream`].
+///
+/// It counts internally the total numbers of subscribers by incrementing the
+/// value every time the [`new`](Self::new) constructor is invoked and
+/// decrementing it when the stream is dropped.
+///
+/// Also the provides a way to track the lagging status of
+/// the subscriber.
+pub struct SubscriberStream<T> {
+    /// The inner stream implementation we want to wrap.
+    inner: BroadcastStream<T>,
+    /// The prometheus label we want to record metrics.
+    label: &'static str,
+    /// The collection of available prometheus metrics we want to record.
+    metrics: Arc<InMemoryStreamMetrics>,
+    /// Tracks if this subscriber is currently lagged.
+    is_lagging: bool,
+}
```

Comment:
Could you elaborate on the reasons we need this new wrapper? How is it supposed to be used by dependent libraries/apps?